compendium/DeclarativeServices/test/TestComponentLifecycle.cpp

```diff
@@ -252,4 +278,101 @@ TEST_F(tServiceComponent, testDelayedComponent_LifeCycle) //DS_TOI_52
   auto sRef2 = ctxt.GetServiceReference<test::Interface2>();
   EXPECT_FALSE(static_cast<bool>(sRef2)) << "Service must not be available after it's dependency is removed";
 }
+
+//To check if error occured during a last function call(Windows Only)
+bool isErrored(const std::string functionName)
+{
+#if defined(US_PLATFORM_WINDOWS)
+    // Retrieve the system error message for the last-error code
+    LPSTR lpDisplayBuf;
+    DWORD dw = GetLastError();
+
+    if (dw == 0)
+        return false;
+
+    FormatMessage(
+        FORMAT_MESSAGE_ALLOCATE_BUFFER |
+        FORMAT_MESSAGE_FROM_SYSTEM |
+        FORMAT_MESSAGE_IGNORE_INSERTS,
+        NULL,
+        dw,
+        MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+        reinterpret_cast<LPTSTR>(&lpDisplayBuf),
+        0, NULL);
+
+    std::cerr << "\nERROR:\n" << functionName << " failed with error " << dw << ": " << lpDisplayBuf << std::endl;
+
+    SetLastError(0);
+    LocalFree(lpDisplayBuf);
+    return true;
+#else
+    std::cerr<<"isErrored("<<functionName<<") will only work for windows";
+    return true;
+#endif
+}
+
+//Function to validate lazy loading of delayed component
+bool isBundleLoaded(const std::string bundleName)
+{
+#if defined(US_PLATFORM_WINDOWS)
+
+    HMODULE hMods[1024];
+    DWORD cbNeeded;
+
+    HANDLE hProcess = GetCurrentProcess();
+
+    EnumProcessModules(hProcess, hMods, sizeof(hMods), &cbNeeded);
+    EXPECT_FALSE(isErrored("EnumProcessModules"));
+
+    EXPECT_GT(sizeof(hMods), cbNeeded) << "Size of array is too small to hold all module handles";
+    if ((sizeof(hMods) < cbNeeded))
+    {
+        return false;
+    }
+
+    TCHAR szModName[MAX_PATH];
+    std::size_t found;
+
+    for (unsigned int i = 0; i < (cbNeeded / sizeof(HMODULE)); i++)
+    {
+        GetModuleFileNameA(hMods[i], szModName, sizeof(szModName) / sizeof(TCHAR));
+        EXPECT_FALSE(isErrored("GetModuleFileNameA"));
```

Comment:
I don't see the point of having `EXPECT_FALSE` here. returning false from this function means the module couldn't be found.  is it worth it to run this code at most `(cbNeeded / sizeof(HMODULE)) -1` times? seems unnecessary.